superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/Options.h

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace Option {
+enum Precision {
+    F16,
+    F32,
+};
+using PrecisionType = Option::Precision;
+} // namespace Option
+
+struct Options {
+    char **begin;
+    char **end;
+
+    /**
+     * @brief Get the char* value of the cmd line argument.
+     * @param option the argument in cmd.
+     * @return char*
+     */
+    char *get_cmd_option(const std::string &option) {
+        char **itr = std::find(begin, end, option);
+        if (itr != end && ++itr != end) {
+            return *itr;
+        }
+        return 0;
+    }
+
+    /**
+     * @brief Get the int type value of cmd line argument.
+     * @param option the cmd line argument.
+     * @return int the int type value of cmd line argument 'option'.
+     */
+    int get_cmd_line_argument_int(const std::string &option, int defaults) {
+        if (char *value = get_cmd_option(option)) {
+            return std::stoi(value);
```

Comment:
>stoi [](http://example.com/codeflow?start=24&length=4)

this function may throw exceptions, so you may add some logic to process it.

Reply:
updated to 
```
    int get_cmd_line_argument_int(const std::string &option, int defaults) {
        if (char *value = get_cmd_option(option)) {
            try
            {
                return std::stoi(value);
            }
            catch(const std::exception& e)
            {
                std::cerr << "Error: Invalid argument - " << option << " should be INT" << '\n';
                exit(1);
            }
        }
        return defaults;
    }
```